### ISSUE TITLE
filesystem encoding and filecontent encoding mixup

### ISSUE THREAD

**Issue by rlindner81** [pyload/pyload#877]
My environment:

> sys.getdefaultencoding()
> 'ascii'
> sys.getfilesystemencoding()
> 'UTF-8'

I traced some crashes to the following problem:
The way fs_encoding/fs_decoding is set up in module.utils it is meant to handle the filesystem encoding, i.e., how filepaths are encoded. Since my filesystem encoding is utf8 no work is required there and both functions are set to lambda x:x. 

However in module.plugins.Plugin.py (line 524) fs_encoding is falsely used to encode the content of a file.

> f.write(fs_encode(res)) 

But encoding the res string (assuming the http header asked for the response to be decoded, decode=True) is really needed, because the default filecontent encoding is ascii. So not encoding it leads to a UnicodeEncodeError.

Long story short, it should be something like

> if decode: res=res.encode('utf8')
> f.write(res)
